Assets/Scripts/UI/DogAdoptionSlot.cs

```diff
@@ -18,13 +21,31 @@ public override void Init(DogDescriptor dog, Sprite dogSprite, Sprite background
 
         priceOrAdoptionStatus = GetComponentInChildren<Text>();
         priceOrAdoptionStatus.text = dog.CostToAdoptStr;
+
+        priceBackgroundImage = images[2];
     }
 
     #endregion
 
     public void Adopt()
     {
-        priceOrAdoptionStatus.text = "Adopted";
+        if (checkAdoption())
+        {
+            dataController.ChangeCoins(-dog.CostToAdopt);
```

Comment:
All of the logic that accesses DataController should be extrapolated into a function called from PPGameController. This class should only need to reference PPGameController for its desired functionality (not PPDataController).